community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/ast/LikeParser.scala

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.compiler.v2_2.ast
+
+import org.parboiled.scala._
+
+//TODO should option be supported? In standard?
+/**
+ * Parser that parses a Like expression into [[ParsedLike]] which contains tokens of [[LikeOp]].
+ *
+ * A Like expression can contain the special characters:
+ *  - % wildcard
+ *  - _ matches exactly one character
+ *  - [XY] optionally match any of the characters
+ */
+case object LikeParser extends Parser {
```

Comment:
Sure, we have few operator and the syntax is simple, but this is the current situation. There is no guarantee it will be so simple forever, moreover having consistent error messages is quite nice. 

On the other hand, we have already a quite monolithic parser, i.e., patterns, expressions, clauses are all parsed by the same parser, so I have a bit of hard time to see the benefit of having a separate parser for this use case.